tianshou/trainer/base.py

```diff
@@ -139,8 +141,7 @@ def __init__(
         step_per_epoch: Optional[int] = None,
         repeat_per_collect: Optional[int] = None,
         episode_per_test: Optional[int] = None,
-        update_per_step: Union[int, float] = 1,
-        update_per_epoch: Optional[int] = None,
```

Comment:
wait is it a compatible change? Pretty dangerous for me. Could you write a test to ensure the behavior is consistent with the previous version?

Reply:
So update_per_step is still there, just no longer an int.

And `update_per_epoch` was unused, so I deleted it. It's breaking in the sense that if a user passed it explicitly as kwarg, it will now break. The way to make it backwards compatible would be to add something like `**kwargs`, which I honestly would rather avoid. I'm not sure what test you have in mind?

Reply:
hm good catch, I just check all `update_per_epoch` usages and it looks like a merge conflict previously that make this argument unused.